zetaclient/evm/evm_client.go

```diff
@@ -107,28 +110,30 @@ var _ interfaces.ChainClient = (*ChainClient)(nil)
 
 // NewEVMChainClient returns a new configuration based on supplied target chain
 func NewEVMChainClient(
+	appContext *appcontext.AppContext,
 	bridge interfaces.ZetaCoreBridger,
 	tss interfaces.TSSSigner,
 	dbpath string,
 	metrics *metricsPkg.Metrics,
-	logger zerolog.Logger,
-	cfg *config.Config,
 	evmCfg config.EVMConfig,
 	ts *metricsPkg.TelemetryServer,
 ) (*ChainClient, error) {
 	ob := ChainClient{
 		ChainMetrics: metricsPkg.NewChainMetrics(evmCfg.Chain.ChainName.String(), metrics),
 		ts:           ts,
 	}
-	chainLogger := logger.With().Str("chain", evmCfg.Chain.ChainName.String()).Logger()
+	chainLogger := appContext.Logger().With().Str("chain", evmCfg.Chain.ChainName.String()).Logger()
 	ob.logger = Log{
 		ChainLogger:          chainLogger,
 		ExternalChainWatcher: chainLogger.With().Str("module", "ExternalChainWatcher").Logger(),
 		WatchGasPrice:        chainLogger.With().Str("module", "WatchGasPrice").Logger(),
 		ObserveOutTx:         chainLogger.With().Str("module", "ObserveOutTx").Logger(),
 	}
-	ob.cfg = cfg
-	ob.params = evmCfg.ChainParams
+	ob.coreContext = appContext.ZetaCoreContext()
+	chainParams, found := appContext.ZetaCoreContext().GetEVMChainParams(evmCfg.Chain.ChainId)
+	if found {
+		ob.chainParams = *chainParams
```

Comment:
The instance of `ChainClient` should not be created, on start, if chain params is missing in zetacore.